modules/integration/tests-integration/lazy-loading-test/src/test/java/org/wso2/appserver/integration/lazy/loading/LazyLoadingBaseTest.java

```diff
@@ -0,0 +1,506 @@
+import org.wso2.carbon.automation.test.utils.http.client.HttpURLConnectionClient;
+import org.wso2.carbon.integration.common.admin.client.ApplicationAdminClient;
+import org.wso2.carbon.integration.common.utils.LoginLogoutClient;
+import org.wso2.carbon.integration.common.utils.mgt.ServerConfigurationManager;
+import org.wso2.carbon.utils.ServerConstants;
+import org.xml.sax.SAXException;
+
+import javax.xml.stream.XMLStreamException;
+import javax.xml.xpath.XPathExpressionException;
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.rmi.RemoteException;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.List;
+
+/**
+ * Base class to hold the common variables and utility methods of Lazy Loading (Ghost deployment) Test cases.
+ * All Lazy Loading (Ghost deployment)  Test  Classes must inherit from this.
+ * Replacing the carbon.xml with Ghost deployment configuration and deployment of the
+ * tenant-info-service web application is implemented in side init() method.
+ */
+public abstract class LazyLoadingBaseTest extends ASIntegrationTest {
+
+
+    protected static final String TENANT_DOMAIN_1_kEY = "tenant1";
+    protected static final String TENANT_DOMAIN_2_KEY = "tenant2";
+    protected static final String SUPER_TENANT_DOMAIN_KEY = "superTenant";
+
+    protected String SUPER_TENANT_DOMAIN;
+    protected String TENANT_DOMAIN_1;
+    protected String TENANT_DOMAIN_2;
+
+    protected WebAppAdminClient webAppAdminClient;
+    protected ServerConfigurationManager serverManager;
+    protected String hostURL;
+    protected static final String CARBON_HOME = System.getProperty(ServerConstants.CARBON_HOME);
+    protected static final String ARTIFACTS_LOCATION = TestConfigurationProvider.getResourceLocation() + File.separator +
+            "artifacts" + File.separator + "AS" + File.separator + "ghost" + File.separator;
+
+    private static final Log log = LogFactory.getLog(LazyLoadingBaseTest.class);
+    private long TENANT_IDLE_TIME;
+    private long WEB_APP_IDLE_TIME;
+    private static final long MAX_THRESHOLD_TIME = 2 * 60 * 1000;
+
+
+    private static final String PRODUCT_GROUP_NAME = "AS";
+    private static final String INSTANCE_NAME = "appServerInstance0001";
+    private static final String USER_KEY = "admin";
+
+    private static final String TENANT_INFO_SERVICE = "lazy-loading-info";
+    private static final String TENANT_INFO_SERVICE_FILE_NAME = TENANT_INFO_SERVICE + ".war";
+    private static final String IS_TENANT_LOADED_METHOD_URL = TENANT_INFO_SERVICE + "/tenant-status";
+    private static final String IS_WEB_APP_LOADED_METHOD_URL = TENANT_INFO_SERVICE + "/webapp-status";
+
+    private static final String TENANT_INFO_SERVICE_ARTIFACT_LOCATION = ARTIFACTS_LOCATION +
+            TENANT_INFO_SERVICE_FILE_NAME;
+
+
+    private static final String TENANT_IDLE_XPATH = "//listenerExtensions/platformExecutionManager/extentionClasses/" +
```

Comment:
as discussed offline, this is the base test class for Ghost deployment test cases.